### ISSUE TITLE
Buildout rather broken

### ISSUE THREAD

**Issue by mjpieters** [collective/buildout.python#75]
From a clean checkout on OS X 10.13.2 I've run into the following problems:

* No `python-3.6-build:darwin` section:

    ```
    While:
      Installing.
      Getting section python-3.6-build.
      Initializing section python-3.6-build.
      Loading input sections for 'python-3.6-build'
    Error: No section named 'python-3.6-build:darwin'
    ```

  I can add 

    ```
    [python-3.6-build:darwin]
    <= python-3.6-build:darwin-elcapitan
    ```

  and continue.

* Buildout is no longer compatible with Python 2.x versions before 2.6:

    ```
    python-2.4: Running
    extends = buildout.cfg
    /Users/mjpieters/Development/Library/buildout.python/parts/opt/bin/python2.4 /Users/mjpieters/Development/Library/buildout.python/src/ez_setup-1.x.py
    /Users/mjpieters/Development/Library/buildout.python/parts/opt/bin/easy_install-2.4 pip==1.1
    /Users/mjpieters/Development/Library/buildout.python/bin/virtualenv-2.4 --system-site-packages /Users/mjpieters/Development/Library/buildout.python/python-2.4
    /Users/mjpieters/Development/Library/buildout.python/python-2.4/bin/pip install --pypi-url=https://pypi.python.org/simple collective.dist
    'import site' failed; use -v for traceback
    Traceback (most recent call last):
      File "/Users/mjpieters/Development/Library/buildout.python/src/ez_setup-1.x.py", line 28, in ?
        from site import USER_SITE
      File "/Users/mjpieters/Development/Library/buildout.python/parts/buildout/site.py", line 146
    extends = buildout.cfg
        with f:
             ^
    SyntaxError: invalid syntax
    'import site' failed; use -v for traceback
    Traceback (most recent call last):
      File "/Users/mjpieters/Development/Library/buildout.python/parts/opt/bin/easy_install-2.4", line 5, in ?
        from pkg_resources import load_entry_point
    ImportError: No module named pkg_resources
    'import site' failed; use -v for traceback
    Traceback (most recent call last):
      File "/Users/mjpieters/Development/Library/buildout.python/parts/virtualenv172/virtualenv.py", line 2429, in ?
        main()
      File "/Users/mjpieters/Development/Library/buildout.python/parts/virtualenv172/virtualenv.py", line 942, in main
        never_download=options.never_download)
      File "/Users/mjpieters/Development/Library/buildout.python/parts/virtualenv172/virtualenv.py", line 1043, in create_environment
        site_packages=site_packages, clear=clear))
      File "/Users/mjpieters/Development/Library/buildout.python/parts/virtualenv172/virtualenv.py", line 1192, in install_python
        import site
      File "/Users/mjpieters/Development/Library/buildout.python/parts/buildout/site.py", line 146
        with f:
             ^
    SyntaxError: invalid syntax
    'import site' failed; use -v for traceback
    Traceback (most recent call last):
      File "/Users/mjpieters/Development/Library/buildout.python/python-2.4/bin/pip", line 5, in ?
        from pkg_resources import load_entry_point
    ImportError: No module named pkg_resources
    python-2.4: Non zero exit code (1) while running command.
    While:
      Updating python-2.4.
    Error: Non zero exit code (1) while running command.
    ```

  for 2.6, the exception changes to:

    ```
      File "/Users/mjpieters/Development/Library/buildout.python/parts/buildout/site.py", line 230, in getuserbase
        from sysconfig import get_config_var
    ImportError: No module named sysconfig
    ```

   I can disable the 2.4, 2.5 and 2.6 parts in a local.cfg file and move on

* We then appear to have a version conflict in setuptools:

    ```
    Installing Setuptools
    Traceback (most recent call last):
      File "setup.py", line 11, in <module>
        import setuptools
      File "/private/var/folders/w0/nl1bwj6163j2pvxswf84xcsjh2pc5g/T/tmpejbFsX/setuptools-33.1.1/setuptools/__init__.py", line 10, in <module>
        from setuptools.extern.six.moves import filter, map
      File "/private/var/folders/w0/nl1bwj6163j2pvxswf84xcsjh2pc5g/T/tmpejbFsX/setuptools-33.1.1/setuptools/extern/__init__.py", line 1, in <module>
        from pkg_resources.extern import VendorImporter
    ImportError: No module named extern
    Something went wrong during the installation.
    See the error message above.
    Traceback (most recent call last):
      File "/Users/mjpieters/Development/Library/buildout.python/parts/opt/bin/easy_install-2.7", line 11, in <module>
        load_entry_point('setuptools==33.1.1', 'console_scripts', 'easy_install-2.7')()
      File "build/bdist.macosx-10.12-intel/egg/pkg_resources.py", line 357, in load_entry_point
      File "build/bdist.macosx-10.12-intel/egg/pkg_resources.py", line 350, in get_distribution
      File "build/bdist.macosx-10.12-intel/egg/pkg_resources.py", line 229, in get_provider
      File "build/bdist.macosx-10.12-intel/egg/pkg_resources.py", line 467, in find
    pkg_resources.VersionConflict: (setuptools 1.4.2 (/Users/mjpieters/Development/Library/buildout.python/eggs/setuptools-1.4.2-py2.7.egg), Requirement.parse('setuptools==33.1.1'))
    Traceback (most recent call last):
      File "/Users/mjpieters/Development/Library/buildout.python/parts/virtualenv/virtualenv.py", line 2332, in <module>
        main()
      File "/Users/mjpieters/Development/Library/buildout.python/parts/virtualenv/virtualenv.py", line 711, in main
        symlink=options.symlink)
      File "/Users/mjpieters/Development/Library/buildout.python/parts/virtualenv/virtualenv.py", line 924, in create_environment
        site_packages=site_packages, clear=clear, symlink=symlink))
      File "/Users/mjpieters/Development/Library/buildout.python/parts/virtualenv/virtualenv.py", line 1144, in install_python
        site_filename_dst = change_prefix(site_filename, home_dir)
      File "/Users/mjpieters/Development/Library/buildout.python/parts/virtualenv/virtualenv.py", line 1035, in change_prefix
        (filename, prefixes)
    AssertionError: Filename /Users/mjpieters/Development/Library/buildout.python/parts/buildout/site.py does not start with any of these prefixes: ['/Users/mjpieters/Development/Library/buildout.python/parts/opt/Extras/lib/python', '/Users/mjpieters/Development/Library/buildout.python/parts/opt', '/Users/mjpieters/Library/Python/2.7/lib/python/site-packages', '/Users/mjpieters/.local/lib/python/2.7/site-packages', '/Users/mjpieters/Library/Python/2.7/site-packages', '/Library/Python/2.7/site-packages']
    python-2.7: Non zero exit code (1) while running command.
    While:
      Installing python-2.7.
    ```

I suspect most of this can be traced back to the new bootstrap script in 82b8c5a84cec (cc: @mauritsvanrees ), but I'll experiment and see what can be done.

**Comment by mjpieters** [comment]
The `python-3.6-build:darwin` dependency was introduced in f3a5909551c by @fschulze; it is obvious in error and I'll correct that one.

**Comment by mauritsvanrees** [comment]
You probably shouldn't use Python2.4 to bootstrap this buildout. Currently for me the biggest reason to use this buildout, is to install a Python2.4 with it, not the other way around.

Technically, the `bootstrap.py` script should be fine, as long as you use `--setup-source` to point to a much older `ez_setup.py` which is Python2.4 compatible.

**Comment by mjpieters** [comment]
> You probably shouldn't use Python2.4 to bootstrap this buildout. Currently for me the biggest reason to use this buildout, is to install a Python2.4 with it, not the other way around.

I'm not bootstrapping with 2.4. I bootstrapped with 2.7. The `site.py` file still applies to the 2.4, 2.5 and 2.6 setups.

The script you replaced was customised to hardcode the 1.4.4 version of buildout. I'll re-introduce that change.

**Comment by mjpieters** [comment]
See 83cd9712654a8d720 specifically; please don't upgrade the bootstrap script unless the site.py issue is resolved.

**Comment by mjpieters** [comment]
Resolved with ed40acef2f429afdbe9cc60d79d62327a5a9f583

**Comment by mauritsvanrees** [comment]
I wasn't aware of problems with `site.py`. Thanks for the fix!
